src/jllvm/vm/native/Lang.cpp

```diff
@@ -15,6 +15,59 @@
 
 #include <llvm/Support/Endian.h>
 
+jllvm::Object* jllvm::lang::ObjectModel::clone()
```

Comment:
As admittedly very confusing as it is, this should probably return `ObjectInterface*` rather than `Object*` which then also simplifies the code below.  The former is more equal to Java's object as it is actually a proper "superclass" of all mirrored Java Objects in C++.

We might want to reevaluate whether we still need `Object` and if we do, maybe rename it to `ObjectInstance` to make it clear that its main use is for creating and dealing with `Object` instances and not subclasses of it. This is definitely a bit of a mess that we created :sweat_smile:
Either way, out of scope for this PR so lets just change it to `ObjectInterface` for now.